CONTRIBUTING.md

```diff
@@ -18,7 +18,119 @@ These are general and loose rules components in UI Kit should strive to fulfill.
 - Documentation of prop-types and usage patterns
 - Sensible defaults for props which are not a hard requirement
 
-### How to add a component to UI Kit.
+## Development
+
+### Testing
+
+We split our testing in three parts: functionality, visual and the final bundle. We use separate tools for these, and we try to avoid overlap.
+
+#### Functionality testing
+
+- write tests from a user perspective, not from a dev perspective
+- we test using `react-testing-library`
+  - we have `test-utils` as a thin wrapper which sets up the context of tested components
+  - we avoid `beforeEach` as it leads to test coupling
+  - an `it` can describe the flow of a user
+  - it is okay to have many `expect` statements in an `it`
+  - code comments can help explain the `expect`s better
+- avoid testing for visuals (classnames or the DOM structure)
+  - the user does not care about the DOM structure or the classnames themselves
+  - use visual testing instead (see below)
+- if mocking is required, try to do it at the level the user interacts with the component
+  - Examples
+    - good: mocking network responses
+    - bad: mocking a fetching component
+  - Try to keep mocking minimal. The less we mock, and the later we mock, the more our tests will cover and the more confidence we gain that our components work
+
+#### Visual testing
+
```

Comment:
maybe we should add the same as in Bundle Testing and explain that they need to run `build`?